scripts/tuple_tree_generator/tuple_tree_generator/schema/schema.py

```diff
@@ -90,6 +91,40 @@ def _resolve_references(self):
         for t in self.definitions.values():
             t.resolve_references(self)
 
+    def _generate_kinds(self):
+        children = {}
+        for key, value in self.definitions.items():
+            if isinstance(value, StructDefinition) and value._inherits is not None:
+                parent = self.get_definition_for(value._inherits)
+                if parent.fullname in children:
+                    children[parent.fullname][1].append(value)
+                else:
+                    children[parent.fullname] = (
+                        parent,
+                        [
+                            value,
+                        ],
+                    )
```

Comment:
Does the dict need to be `parent_name: str -> parent: Definition`? I don't see parent_name used below and IIRC `get_definition_for` returns always the same instance for a given type name.
I'm not sure `Definition` is hashable at the moment, but in case it's not implementing `__hash__` as `return id(self)` should be good enough.
Also, I would suggest using a `defaultdict(list)` so you can remove the `if parent.fullname in children` test and just do `children[parent].append(value)`.

Reply:
Changed, this is way more elegant